src/internal/pjsdb/queue_iterator.go

```diff
@@ -0,0 +1,74 @@
+package pjsdb
+
+import (
+	"context"
+
+	"github.com/jmoiron/sqlx"
+
+	"github.com/pachyderm/pachyderm/v2/src/internal/errors"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pachsql"
+	"github.com/pachyderm/pachyderm/v2/src/internal/pctx"
+	"github.com/pachyderm/pachyderm/v2/src/internal/stream"
+)
+
+// TODO(Fahad): add queue filter once filter is designed.
+type IterateQueuesRequest struct {
+	IteratorConfiguration
+}
+
+// QueuesIterator implements stream.Iterator[T] on Queue objects following the
+// pattern set by the pfsdb package.
+// Iterators are the underlying type used by list-style crud operations.
+type QueuesIterator struct {
```

Comment:
Why `QueuesIterator` rather than `QueueIterator`?

Reply:
Because it iterates over _queues_